src/main/primitives/Point.java

```diff
@@ -40,45 +18,27 @@
 	 * @param z The z-coordinate.
 	 */
 	public Point(double x, double y, double z) {
-		this.x = x;
-		this.y = y;
-		this.z = z;
+		super(x, y, z);
 	}
 
 	/**
-	 * Creates a new {@link Point} which is a transformation of this {@link Point} by applying the given transformation
-	 * to each of the coordinates.
+	 * This constructor accepts a {@link Triple} and constructs the appropriate {@link Point} with those values.
 	 *
-	 * @param transformation A function which receives two coordinates and returns another coordinate.
-	 * @param aux An auxiliary {@link Point} whose corresponding coordinate may (or may not) be used in the
-	 *        transformation function in order to calculate each of the new coordinates.
-	 * @return The {@link Point} made up of applying the transformation to each of the three coordinates.
+	 * @param triple The {@link Triple} with the coordinates which this {@link Point} has.
+	 * @throws ZeroVectorException if this {@link Vector} is the zero vector.
 	 */
-	public Point transform(DoubleBinaryOperator transformation, Point aux) {
-		return new Point(transformation.applyAsDouble(x, aux.x),
-			transformation.applyAsDouble(y, aux.y), transformation.applyAsDouble(z, aux.z));
-	}
-
-	/**
-	 * Similar to {@link #transform(DoubleBinaryOperator, Point)} but does not require an auxiliary {@link Point}, since
-	 * the transformation when called in this way does not depend on a second coordinate.
-	 *
-	 * @param transformation A function which receives a single coordinate and returns another coordinate.
-	 * @return The {@link Point} made up of applying the transformation to each of the three coordinates.
-	 */
-	public Point transform(DoubleUnaryOperator transformation) {
-		return new Point(transformation.applyAsDouble(x), transformation.applyAsDouble(y),
-			transformation.applyAsDouble(z));
+	public Point(Triple triple) {
+		this(triple.x, triple.y, triple.z);
 	}
 
 	/**
 	 * Adds a {@link Vector} to this {@link Point} and returns the resulting {@link Point}.
 	 *
-	 * @param v The {@link Vector} to add to this {@link Point}.
+	 * @param vector The {@link Vector} to add to this {@link Point}.
 	 * @return The {@link Point} resulting from adding the {@link Vector} to this {@link Point}.
 	 */
-	public Point add(Vector v) {
-		return transform((base, aux) -> base + aux, v.head);
+	public Point add(Vector vector) {
+		return new Point(transform(Double::sum, vector));
```

Comment:
weird syntax bro

Reply:
It's a function pointer to an existing function rather than reimplement it with a lambda